src/authorization/OwnershipAuthorizer.ts

```diff
@@ -0,0 +1,41 @@
+import type { AccountSettings, AccountStore } from '../identity/interaction/email-password/storage/AccountStore';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { Authorization } from './Authorization';
+import type { AuthorizerArgs } from './Authorizer';
+import { Authorizer } from './Authorizer';
+
+/**
+ * Authorizer used to make sure that owners always will be able to change the permissions on data stored in their pod.
+ */
+export class OwnershipAuthorizer extends Authorizer {
+  private readonly accountStore: AccountStore;
+
+  public constructor(accountStore: AccountStore) {
+    super();
+    this.accountStore = accountStore;
+  }
+
+  public async canHandle({ credentials, identifier, permissions }: AuthorizerArgs): Promise<void> {
+    if (Object.entries(permissions).some(([ name, value ]): boolean => value && name !== 'control')) {
+      throw new NotImplementedHttpError('Only control permissions are supported.');
+    }
+    if (!credentials.webId) {
+      throw new NotImplementedHttpError('Only authenticated requests are supported.');
+    }
+    let settings: AccountSettings;
+    try {
+      settings = await this.accountStore.getSettings(credentials.webId);
+    } catch {
+      throw new NotImplementedHttpError('Only requests by registered WebIDs are supported.');
+    }
+    if (!settings.podBaseUrl || !identifier.path.startsWith(settings.podBaseUrl)) {
+      throw new NotImplementedHttpError('Only requests targeting the pod registered to this WebID are supported.');
+    }
+  }
+
+  public async handle(): Promise<Authorization> {
+    // If all checks in the canHandle function pass permission is always granted
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    return { addMetadata(): void {} };
+  }
```

Comment:
I'm a bit surprised by the `canHandle`/`handle` division: I would expect `handle` to check whether the requesting agent is the owner of this pod. (So probably a reverse check.)

Reply:
The reason for this is because of some of the issues that you raise. It is written as it is so in the case this check fails (it's not the owner, the request is not for control permissions, etc.) the next handler will be called (the main acl authorizer).

Reply:
Agreed.

Reply:
To be precise: I agree with the code as-is, namely that it only has something to say in the specific case where an owner asks for `Control` access.